test/e2e/cert_manager_deployment_test.go

```diff
@@ -1,27 +1,53 @@
 package e2e
 
 import (
+	"context"
+	"embed"
+	"path/filepath"
 	"testing"
+	"time"
+
+	_ "embed"
 
 	"github.com/stretchr/testify/require"
-	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
-	"k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/cert-manager-operator/test/library"
 )
 
-func NewClientConfigForTest(t *testing.T) *rest.Config {
-	loader := clientcmd.NewDefaultClientConfigLoadingRules()
-	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loader, &clientcmd.ConfigOverrides{ClusterInfo: api.Cluster{InsecureSkipTLSVerify: true}})
-	config, err := clientConfig.ClientConfig()
-	if err == nil {
-		t.Logf("Found configuration for host %v.\n", config.Host)
-	}
+const (
+	PollInterval = time.Second
+	TestTimeout  = 10 * time.Minute
+)
 
-	require.NoError(t, err)
-	return config
-}
+//go:embed testdata/*
+var testassets embed.FS
 
-func TestDeployment(t *testing.T) {
-	kubeConfig := NewClientConfigForTest(t)
-	t.Logf("This will be updated once the Prow tests boot up fine. KubeConfig %v\n", kubeConfig)
+func TestSelfSignedCerts(t *testing.T) {
+	kubeClient, dynamicKubeConfig := library.NewClientsConfigForTest(t)
+
+	library.CreateFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "namespace.yaml"), kubeClient, dynamicKubeConfig)
+	defer library.DeleteFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "namespace.yaml"), kubeClient, dynamicKubeConfig)
+	library.CreateFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "cluster_issuer.yaml"), kubeClient, dynamicKubeConfig)
+	defer library.DeleteFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "cluster_issuer.yaml"), kubeClient, dynamicKubeConfig)
+	library.CreateFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "issuer.yaml"), kubeClient, dynamicKubeConfig)
+	defer library.DeleteFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "issuer.yaml"), kubeClient, dynamicKubeConfig)
+	library.CreateFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "certificate.yaml"), kubeClient, dynamicKubeConfig)
+	defer library.CreateFromFile(t, testassets.ReadFile, filepath.Join("testdata", "self_signed", "certificate.yaml"), kubeClient, dynamicKubeConfig)
+
+	err := wait.PollImmediate(PollInterval, TestTimeout, func() (bool, error) {
+		secret, err := kubeClient.CoreV1().Secrets("sandbox").Get(context.TODO(), "root-secret", metav1.GetOptions{})
```

Comment:
Let's wire those context.TODO()s here and elsewhere, since this is net-new code.

Reply:
Done - I converted this part into a struct and implemented this in a more OO way.